src/librustc/middle/const_eval.rs

```diff
@@ -357,11 +359,15 @@ pub enum ErrKind {
     NegateOnBinary,
     NegateOnStruct,
     NegateOnTuple,
+    NegateOnArray,
+    NegateOnRepeat,
     NotOnFloat,
     NotOnString,
     NotOnBinary,
     NotOnStruct,
     NotOnTuple,
+    NotOnArray,
+    NotOnRepeat,
```

Comment:
(I'm okay with not trying to fold that particular change into this PR -- i.e. switching to a nested enum might make sense but also seems like it is a broader refactoring that we could develop later, no?)